chain/consensus/mir/state_manager.go

```diff
@@ -201,6 +218,8 @@ func (sm *StateManager) RestoreState(checkpoint *checkpoint.StableCheckpoint) er
 func (sm *StateManager) ApplyTXs(txs []*requestpb.Request) error {
 	var mirMsgs []Message
 
+	sm.height++
```

Comment:
Are we sure the update of `sm.Height` is correct? This may be the reason for the snapshot or `WaitForBlock` to get stuck waiting for some block number

Reply:
@matejpavlovic Could you please answer this question?

Reply:
I am sure that it needs to be incremented here, if this is what you ask.

Whether there are any bugs related to this value around reconfiguration, state transfer, etc... I haven't found any so far, but can't guarantee there are none.